Automattic/pocket-casts-android
# [Paywall Experiments] Implement ExPlat

## Description
- The goal of this PR is to replace the AB test control for the paywall experiments from Firebase to Explat
- We are going to keep this behind feature flag: p1726143653353199/1725977109.174469-slack-C07G1MH890E 
- To be able to use Explat you need to have `EXPLAT_EXPERIMENT` feature flag enabled
- See for more context: paqN3M-1fz-p2#comment-1310
- See the experiment: 21899-explat-experiment
- See ExPlat SDK: https://github.com/Automattic/Automattic-Tracks-Android/tree/trunk/experimentation#readme
- See iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/1969


Fixes #2870

## Testing Instructions
- Mandatory to have the `EXPLAT_EXPERIMENT` FF enabled

I am currently trying to figure out how to change from Control to Treatment to see both flows for the AB testing. You can see the slack thread here: p1726145194421799-slack-C07DLM97HRQ


## Checklist
 
#### I have tested any UI changes...
